tests/Unit/DataStructures/Test_TaggedContainers.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/TaggedContainers.hpp"
+#include "DataStructures/Tags.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+namespace Tags {
+struct MyDouble : db::SimpleTag {
+  using type = double;
+};
+struct MyScalar : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+struct MyDouble2 : db::SimpleTag {
+  using type = double;
+};
+}  // namespace Tags
+
+// Test that desired tags are retrieved.
+void test_tagged_containers() {
+  auto box = db::create<db::AddSimpleTags<Tags::MyDouble>>(1.2);
+  tuples::TaggedTuple<Tags::MyDouble2> tuple(4.4);
+  Variables<tmpl::list<Tags::MyScalar>> vars(10, 3.2);
+
+  auto retrieved_value1 = get<Tags::MyDouble>(box);
```

Comment:
Refactor these into `CHECK(get<...>(...)` and then test with all combinations of 1, 2, 3 args that won't result in a compiler error to make sure you get what is expected. There are a lot of code paths in the hpp that need to be hit, with a bunch of special cases.